src/util/system.h

```diff
@@ -35,6 +35,10 @@
 
 #include <boost/thread/condition_variable.hpp> // for boost::thread_interrupted
 
+#ifndef WIN32
+void SetDefaultUmask();
```

Comment:
Wouldn't `SetupEnvironment` be the right place for this? Or is that too early?

Reply:
Currently, `SetDefaultUmask()` is called just before the first possible disk write.

`SetupEnvironment()` seems too broad, as it is called in utilities too. And, yes, I think it is too early. But I could be wrong ;)

Reply:
But does that matter? What would be the consequences of setting it too early? Why wouldn't we want to set the umask for the utilties?

Reply:
> Wouldn't `SetupEnvironment` be the right place for this?

Agree.